src/main/java/org/ensime/maven/plugins/ensime/EnsimeConfigGenerator.java

```diff
@@ -685,36 +685,39 @@ private void write(final String content, final File out) {
     } catch(IOException ioex) {}
   }
 
-  /**
-   * Generates configurations.
-   */
-  public void generate(final File out) {
-
-    String projectDir =
-      project.getBasedir().toPath().toAbsolutePath().toString();
+  public EnsimeConfig generateConfig() {
```

Comment:
I see you split this up instead of making others public, which is okay. But limit the exposure by making it `projected`, I see no point of it being available publicly.

Reply:
Good point! I'll do that. Thanks!